substratools/algo.py

```diff
@@ -99,18 +100,16 @@ def execute(self, method_name: str, rank: int = 0, fake_data: bool = False, n_fa
         # load inputs
         inputs = deepcopy(self._workspace.task_inputs)
 
-        inputs.update({"rank": rank})
+        task_properties = {"rank": rank}
 
         # load data from opener
         if self._opener_wrapper:
-            X = self._opener_wrapper.get_X(fake_data, n_fake_samples)
-            y = self._opener_wrapper.get_y(fake_data, n_fake_samples)
+            loaded_datasamples = self._opener_wrapper.get_data(fake_data, n_fake_samples)
 
             if fake_data:
                 logger.info("Using fake data with %i fake samples." % int(n_fake_samples))
 
-            inputs.update({"X": X})
-            inputs.update({"y": y})
+            inputs.update({TASK_IO_DATASAMPLES: loaded_datasamples})
```

Comment:
I know this is not supposed to happen, you can add an assert that the `TASK_IO_DATASAMPLES` key is not already present
assert because it would not be because of user input but an error in how the inputs are passed to the task

Reply:
done